docs: include mentions of pre-aggregations with `scheduledRefreshContexts` (#1971)
cube-js/cube.js: docs/content/Configuration/Multitenancy-Setup.md

- `preAggregationsSchema`
- `queryTransformer`

All of the above options are functions, which you provide to Cube.js in the
[`cube.js` configuration file][ref-config]. The functions accept one argument -
a context object, which has a [`securityContext`][ref-config-security-ctx]

> property where you can provide all the necessary data to identify a user i.e.

Reviewer: ```suggestion
property where you can provide all the necessary data to identify a user e.g., 
```

organization, app, etc. By default, the
[`securityContext`][ref-config-security-ctx] is defined by [Cube.js API
Token][ref-security].

There're several multitenancy setup scenarios that can be achieved by using
combinations of these configuration options.

### Multitenancy vs Multiple Data Sources

In cases where your Cube.js schema is spread across multiple different databases
you may consider using the [`dataSource` cube property][ref-cube-datasource]
instead of multitenancy. Multitenancy designed for cases where you need to serve
different datasets for multiple users or tenants which aren't related to each

> other. On other hand multiple data sources can be used for a scenario where

Reviewer: ```suggestion
other. On the other hand, multiple data sources can be used for a scenario where
```

> users need to access same data but from different databases. Multitenancy and

Reviewer: ```suggestion
users need to access the same data but from different databases. Multitenancy and
```

      return new AthenaDriver();
    } else if (dataSource === 'googleAnalytics') {
      return new BigQueryDriver();
    } else if (dataSource === 'financials') {
      return new PostgresDriver({
        database: 'financials',
        host: 'financials-db.acme.com',
        user: process.env.FINANCIALS_DB_USER,
        password: process.env.FINANCIALS_DB_PASS,
      });
    } else {
      return new PostgresDriver();
    }
  },
};
```

### Security Context vs Multitenant Compile Context

As a rule of thumb, [`SECURITY_CONTEXT`][ref-cube-security-ctx] should be used
in scenarios when you want to define row-level security within the same database

> for different users of such database. For example to separate access of two

Reviewer: ```suggestion
for different users of such database. For example, to separate access of two
```

> e-commerce administrators who work on different product categories within same

Reviewer: ```suggestion
e-commerce administrators who work on different product categories within the same
```

e-commerce store.

```javascript
cube(`Products`, {
  sql: `select * from products where ${SECURITY_CONTEXT.categoryId.filter(
    'categoryId'
  )}`,
});
```

> On other hand Multitenant [`COMPILE_CONTEXT`][ref-cube-security-ctx] should be

Reviewer: ```suggestion
On the other hand, Multitenant [`COMPILE_CONTEXT`][ref-cube-security-ctx] should be
```

> used when users in fact access different databases. For example if you provide

Reviewer: ```suggestion
used when users in fact access different databases. For example, if you provide
```

> SaaS ecommerce hosting and each of your customers has separate database then

Reviewer: ```suggestion
SaaS ecommerce hosting and each of your customers have a separate database, then
```

  sql: `select * from ${tenantId}.products`,
});
```

### Security Context vs queryTransformer

[`SECURITY_CONTEXT`][ref-cube-security-ctx] is great for use cases where you
want to get explicit control over filtering of underlying data seen by users.
However for use cases where you want to reuse pre-aggregation tables for
different users or even tenants, using
[`queryTransformer`][ref-config-query-transform] is a much better choice.
[`queryTransformer`][ref-config-query-transform] is also very convenient way of
enforcing row level security by means of join logic defined in your cubes
instead of embedding [`SECURITY_CONTEXT`][ref-cube-security-ctx] filtering
boilerplate into each cube. Together with
[`contextToOrchestratorId`][ref-config-ctx-to-orch-id], this allows defining
both row-level security filtering as well as reusing the same pre-aggregation
set for each tenant.

## Same DB Instance with per Tenant Row Level Security

Per tenant row-level security can be achieved by configuring
[`queryTransformer`][ref-config-query-transform], which adds a tenant identifier
filter to the original query. It uses the
[`securityContext`][ref-config-security-ctx] to determine which tenant is

> requesting data. This way, every tenant starts to see their own data however

Reviewer: ```suggestion
requesting data. This way, every tenant starts to see their own data. However,
```

};
```

Next, we can use [`driverFactory`][ref-config-driverfactory] to dynamically
select database, based on `appId` and `userId`.

**cube.js:**

```javascript
const PostgresDriver = require('@cubejs-backend/postgres-driver');

module.exports = {
  contextToAppId: ({ securityContext }) =>
    `CUBEJS_APP_${securityContext.appId}_${securityContext.userId}`,
  driverFactory: ({ securityContext }) =>
    new PostgresDriver({
      database: `my_app_${securityContext.appId}_${securityContext.userId}`,
    }),
};
```

## Same DB Instance with per Tenant Pre-Aggregations

To support per-tenant pre-aggregation of data within the same database instance,
you should configure the [`preAggregationsSchema`][ref-config-preagg-schema]
option in your `cube.js` configuration file. You should use also
[`securityContext`][ref-config-security-ctx] to determine which tenant is
requesting data.

**cube.js:**

```javascript
module.exports = {
  contextToAppId: ({ securityContext }) =>
    `CUBEJS_APP_${securityContext.userId}`,
  preAggregationsSchema: ({ securityContext }) =>
    `pre_aggregations_${securityContext.userId}`,
};
```

## Multiple Schema and Drivers

> What if for application with ID 3 data is stored not in Postgres, but in

Reviewer: ```suggestion
What if for application with ID 3, the data is stored not in Postgres, but in
```
